File: codeMRI.ASTService/tests/fixtures/python/simple.py
# Simple Python file for testing

def calculate_sum(a, b):
    """Add two numbers and return the result."""
    return a + b

class Calculator:
    def __init__(self):
        self.value = 0
    
    def add(self, x):
        """Add a number to the current value."""
        self.value += x
        return self.value

if __name__ == "__main__":
    calc = Calculator()
    result = calc.add(5)
    print(f"Result: {result}")

File: codeMRI.ASTService/tests/fixtures/typescript/simple.ts
// Simple TypeScript file for testing

interface ICalculator {
    value: number;
    add(x: number): number;
}

class Calculator implements ICalculator {
    value: number = 0;

    add(x: number): number {
        this.value += x;
        return this.value;
    }

    static create(): Calculator {
        return new Calculator();
    }
}

// Main entry point
function main(): void {
    const calc = Calculator.create();
    const result = calc.add(5);
    console.log(`Result: ${result}`);
}

// Export for module usage
export { Calculator, ICalculator };
